### DISCUSSION TITLE
Adding custom accessories to the product / checkout - like engravings on bottles

### DISCUSSION THREAD

**Discussion by TurkaljGoran** [saleor/saleor#15173] [Ideas]
Hello Saleor team, 

we're currently trying to add an "engraving" to a product - a bottle of liquer.
The engraving should cost 8 euros.

Is there a way to achieve the following feature in Saleor: 

1. User adds a bottle of liquer (100 euros) to their checkout and choose the engraving text to be applied to the bottle.
2. Engraving costs 8 euros
3. User applies voucher (or any other discount) to the checkout
4.  Checkout / product gets discounted but the engraving price of 8 euros is left out of the discount. 

Currently in Saleor v3.16.6 it seems impossible to achieve this because if we "add" the engraving price to the checkout by manipulating product unit price, the discount will still get applied in the end: 

1. product is 100 euros
2. engraving costing 8 euros is added
3. we have to mainpulate product unit price in checkout to set it to 108 euros
4. discount is applied
5. Saleor applies discount to 108 euros

We want the discount to be applied to 100 euros (initial product price) and 8 euros to be left out of the discount calculation. 
Is this possible in any version of Saleor or maybe planned in future releases?

Any hint, info or feedback is greatly appreciated!

**Comment by timuric** [comment]
Hey @TurkaljGoran you can use metadata to store engraving info, for the engraving service you can create independent product (which is actually a service) which would cost 8 euros. To connect the engraving to a product you can use store product id in metadata. Hope that makes sense

**Reply by TurkaljGoran** [reply]
Thank you @timuric  we thought about going with that exact approach. 

We'd need the calculation of the shipping methods for the checkout to "ignore" the engraving when determining if free shipping method threshold is reached. 

For example if a product costs 68 euros, and threshold for free shipping is 70 euros, by adding engraving product with 8 euros price it would cause Saleor to calculate 76 euros for the checkout and therefore apply free shipping shipping method. 

 Is there any easy way to "exclude a particular product's price when shipping methods for the checkout are calculated?
 
 In the dashboard we can see that a shipping method can have excluded products but that means that the shipping method itself cannot be used for the checkout which has the excluded product. 
 
 Our use case would be "hey, this engraving product's price should not interfere with shipping method calculation at all".

**Reply by timuric** [reply]
You would have to use webhooks to calculate shipping price dynamically https://docs.saleor.io/docs/3.x/developer/extending/webhooks/synchronous-events/shipping
